src/common/illuminants.h

```diff
@@ -441,14 +441,14 @@ static int find_temperature_from_raw_coeffs(const dt_image_t *img, const dt_alig
         XYZ_to_CAM[k][i] = img->adobe_XYZ_to_CAM[k][i];
   }
 
-  if(isnan(XYZ_to_CAM[0][0])) return FALSE;
+  if(!(XYZ_to_CAM[0][0] < FLT_MAX)) return FALSE; // fail if first element is NAN
```

Comment:
Are you completely sure, that compiler won't transform this to `if(XYZ_to_CAM[0][0] >= FLT_MAX)` during optimization stage (which I suspect is valid, since we declared no NaNs or Infs), which will give the exact opposite result from the one you're expecting?

Reply:
Yes, it's theoretically possible but in practice I see compilers merely reversing the flag test for the branch when using !.  The whole use of NaNs as flags needs to be revisited, since eliminating them should allow using -ffast-math globally and likely yield a speedup in numerous modules.